cmd/pmgr/main.go

```diff
@@ -1,11 +1,178 @@
 package main
 
 import (
+	"bytes"
+	"flag"
 	"fmt"
-
-	"github.com/hftamer/go-training/pkg/pmgr"
+	"golang.org/x/crypto/bcrypt"
+	"log"
+	"os"
+	"strings"
 )
 
+type userMap map[string]string
```

Comment:
Instead of a typedef like this, consider making a `struct` that wraps the `map` and exposes methods to manipulate it. In addition, we can use [struct tags](https://www.digitalocean.com/community/tutorials/how-to-use-struct-tags-in-go) to mark up the type as JSON for easy (de)serialization:

```
type Account struct {
    name string `json:"name"`
    password string `json:"password"`
}
```